packages/common/src/store/buy-crypto/sagas.ts

```diff
@@ -118,6 +127,165 @@ function* getBuyCryptoRemoteConfig(mint: MintName) {
   }
 }
 
+function* swapSolForCrypto({
+  wallet,
+  mint,
+  feePayer,
+  quoteResponse,
+  userbank
+}: {
+  wallet: Keypair
+  mint: MintName
+  feePayer: PublicKey
+  userbank: PublicKey
+  quoteResponse: Awaited<ReturnType<typeof jupiterInstance.quoteGet>>
+}) {
+  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+  // Create a memo
+  // See: https://github.com/solana-labs/solana-program-library/blob/d6297495ea4dcc1bd48f3efdd6e3bbdaef25a495/memo/js/src/index.ts#L27
+  const memoInstruction = new TransactionInstruction({
+    keys: [
+      {
+        pubkey: wallet.publicKey,
+        isSigner: true,
+        isWritable: true
+      }
+    ],
+    programId: MEMO_PROGRAM_ID,
+    data: Buffer.from(`In-App $${mint.toUpperCase()} Purchase: Link by Stripe`)
+  })
+
+  // Create a temp wSOL account
+  const walletSolTokenAccount = getAssociatedTokenAddressSync(
+    NATIVE_MINT,
+    wallet.publicKey
+  )
+  const createWSOLInstruction =
+    createAssociatedTokenAccountIdempotentInstruction(
+      feePayer, // fee payer
+      walletSolTokenAccount, // account to create
+      wallet.publicKey, // owner
+      NATIVE_MINT // mint
+    )
+
+  // Transfer the SOL to the wSOL account
+  const transferWSOLInstruction = SystemProgram.transfer({
+    fromPubkey: wallet.publicKey,
+    toPubkey: walletSolTokenAccount,
+    lamports:
+      quoteResponse.swapMode === 'ExactIn'
```

Comment:
the rest of this method is just copy/pasted from before, but this is one main difference to account for the different SwapMode